internal/mapper/attrmapper/data_source_attributes.go

```diff
@@ -20,6 +21,7 @@ type DataSourceAttribute interface {
 
 type DataSourceNestedAttribute interface {
 	ApplyNestedOverride([]string, explorer.Override) (DataSourceAttribute, error)
+	NestedMerge([]string, DataSourceAttribute, schema.ComputedOptionalRequired) (DataSourceAttribute, error)
```

Comment:
Maybe a more explicit name like `NestedMergeAtPath`? (I'm not particularly fond of how our original code is named 😆)